File: files/CH-199.py
import pandas as pd

path = "CH-199 Combining the columns.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="H", skiprows=1, nrows=6)

def combine_columns(row):
    first_name, middle_name, last_name, pattern = row
    order = list(map(int, pattern.split(',')))
    names = [first_name, middle_name, last_name]
    return ' '.join([names[i-1] for i in order])

input['Custom Format'] = input.apply(combine_columns, axis=1)
result = input['Custom Format']

print(all(input['Custom Format'] == test['Custom Format'])) # True

File: files/CH-199.R
library(tidyverse)
library(readxl)

path = "files/CH-199 Combining the columns.xlsx"
input = read_excel(path, range = "B2:E7")
test  = read_excel(path, range = "H2:H7")

result = input %>%
  mutate(result = pmap_chr(
    list(`First Name`, `Middle Name`, `Last Name`, Pattern),
    function(a, b, c, ord_str) {
      order <- as.integer(str_split(ord_str, ",", simplify = TRUE))
      paste(c(a, b, c)[order], collapse = " ")
    }
  )) %>%
  select(result)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE